### DISCUSSION TITLE
Lint or format rule to get rid of "irregular whitespaces"

### DISCUSSION THREAD

**Discussion by QuentinFchx** [astral-sh/ruff#14674] [Ideas]
Hi,

**Context**
Very often on MacOS, when typing swiftly, I inadvertently insert non-breaking spaces in my code.
Especially after pipes `|` (used for union types) : the keyboard combination for the pipe is (<kbd>⌥ + Shift + L</kbd>), I don't release <kbd>⌥</kbd> quickly enough and then <kbd>⌥ + Space</kbd> insert a non-breaking space (` ` / U+00A0) which is causing SyntaxErrors.
I guess I'm not [the only one](https://superuser.com/questions/416022/hidden-characters-inserted-after-pipe-followed-by-a-space)... 😅 

More generally, I guess these space-like characters in the code are usually not wanted and _hard_ to detect (they look like regular spaces), if ruff could help with getting rid of those it would be awesome.

**Idea**
Would it be in the scope of ruff to add a (formatting ?) rule that would convert such chars into normal whitespaces, as long as they're not part of a string or a comment?
Other lint tools implement a similar behaviour, such as eslint with the [no-irregular-whitespace rule](https://eslint.org/docs/latest/rules/no-irregular-whitespace)

Thank you in advance :)

**Comment by MichaReiser** [comment]
I think having this part of Ruff makes sense, especially in cases where it results in a syntax error because of it.
